src/offline/cable_output.F90

```diff
@@ -725,66 +747,66 @@ SUBROUTINE open_output_file(dels, soil, veg, bgc, rough)
        out%Qscrn = 0.0 ! initialise
     END IF
-    IF(output%veg .OR. output%VegT) THEN
+    IF(output%VegT) THEN
        CALL define_ovar(ncid_out, ovid%VegT, 'VegT', 'K',                      &
             'Average vegetation temperature', patchout%VegT,       &
             'dummy', xID, yID, zID, landID, patchID, tID)
        ALLOCATE(out%VegT(mp))
        out%VegT = 0.0 ! initialise
     END IF
-    IF(output%veg .OR. output%CanT) THEN
+    IF(output%CanT) THEN
        CALL define_ovar(ncid_out, ovid%CanT, 'CanT', 'K', &
             'Within-canopy temperature', patchout%CanT, &
             'dummy', xID, yID, zID, landID, patchID, tID)
        ALLOCATE(out%CanT(mp))
        out%CanT = 0.0 ! initialise
     END IF
-    IF(output%veg .OR. output%Fwsoil) THEN
+    IF(output%Fwsoil) THEN
        CALL define_ovar(ncid_out, ovid%Fwsoil, 'Fwsoil', '[-]', &
             'soil moisture modifier to stomatal conductance', patchout%Fwsoil, &
             'dummy', xID, yID, zID, landID, patchID, tID)
        ALLOCATE(out%Fwsoil(mp))
        out%Fwsoil = 0.0 ! initialise
     END IF
-    IF(output%veg .OR. output%CanopInt) THEN
+    IF(output%CanopInt) THEN
        CALL define_ovar(ncid_out, ovid%CanopInt, 'CanopInt', 'kg/m^2',         &
             'Canopy intercepted water storage', patchout%CanopInt, &
             'dummy', xID, yID, zID, landID, patchID, tID)
        ALLOCATE(out%CanopInt(mp))
        out%CanopInt = 0.0 ! initialise
     END IF
-    IF(output%veg .OR. output%LAI) THEN
+    IF(output%LAI) THEN
        CALL define_ovar(ncid_out, ovid%LAI, 'LAI', '-',                        &
             'Leaf area index', patchout%LAI, 'dummy', xID,         &
             yID, zID, landID, patchID, tID)
        ALLOCATE(out%LAI(mp))
        out%LAI = 0.0 ! initialise
     END IF
     ! Define balance variables in output file and allocate temp output vars:
-    IF(output%balances .OR. output%Ebal) THEN
+    IF(output%Ebal) THEN
        CALL define_ovar(ncid_out, ovid%Ebal, 'Ebal', 'W/m^2',                  &
             'Cumulative energy imbalance', patchout%Ebal,          &
             'dummy', xID, yID, zID, landID, patchID, tID)
        ALLOCATE(out%Ebal(mp))
        out%Ebal = 0.0 ! initialise
     END IF
-    IF(output%balances .OR. output%Wbal) THEN
+    IF(output%Wbal) THEN
        CALL define_ovar(ncid_out, ovid%Wbal, 'Wbal', 'kg/m^2',                 &
             'Cumulative water imbalance', patchout%Wbal,           &
             'dummy', xID, yID, zID, landID, patchID, tID)
        ALLOCATE(out%Wbal(mp))
        out%Wbal = 0.0 ! initialise
     END IF
     ! Define carbon variables in output file and allocate temp output vars:
-    IF(output%carbon .OR. output%AutoResp) THEN
+    IF(output%AutoResp) THEN
        CALL define_ovar(ncid_out, ovid%AutoResp, 'AutoResp', 'umol/m^2/s',     &
             'Autotrophic respiration', patchout%AutoResp,          &
             'dummy', xID, yID, zID, landID, patchID, tID)
```

Comment:
Configuration : `AU-Tum_2002-2017_OzFlux_Met_R0_S0_out.nc`

Change: L816 uses the default configuration of range checks to include `StemResp` and `LeafResp` in nc headers if `output%carbon` is `true` (since `AutoResp` is being set in `set_group_output_values`). So the effective change is that L816 and L824 are using the condition `IF(output%casa .OR. output%carbon .OR. output%AutoResp)`. Based on the behaviour by other functions, I think this is the supposed functionality that we want

Reply:
L794 should be corrected to `output%casa .AND. output%AutoResp` to account for the same output file and the desired behaviour of setting group values